app/src/main/java/com/mapbox/maps/testapp/examples/markersandcallouts/ViewAnnotationShowcaseActivity.kt

```diff
@@ -0,0 +1,197 @@
+import com.mapbox.maps.viewannotation.ViewAnnotationManager
+import com.mapbox.maps.viewannotation.viewAnnotationOptions
+
+/**
+ * Example how to add view annotations to the map.
+ *
+ * Specifically view annotations will be associated with marker icons
+ * showcasing how to implement functionality similar to MarkerView from Maps v9.
+ */
+class ViewAnnotationShowcaseActivity : AppCompatActivity(), OnMapClickListener, OnMapLongClickListener {
+
+  private lateinit var mapboxMap: MapboxMap
+  private lateinit var viewAnnotationManager: ViewAnnotationManager
+  private lateinit var style: Style
+  private val pointList = mutableListOf<Feature>()
+  private var markerId = 0
+
+  private var markerWidth = 0
+  private var markerHeight = 0
+
+  private val asyncInflater by lazy { AsyncLayoutInflater(this) }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    val binding = ActivityViewAnnotationShowcaseBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+
+    viewAnnotationManager = binding.mapView.viewAnnotationManager
+
+    val bitmap = BitmapFactory.decodeResource(resources, R.drawable.blue_marker_view)
+    markerWidth = bitmap.width
+    markerHeight = bitmap.height
+
+    mapboxMap = binding.mapView.getMapboxMap().apply {
+      loadStyle(
+        styleExtension = prepareStyle(Style.MAPBOX_STREETS, bitmap)
+      ) {
+        style = it
+        addOnMapClickListener(this@ViewAnnotationShowcaseActivity)
+        addOnMapLongClickListener(this@ViewAnnotationShowcaseActivity)
+        binding.fabStyleToggle.setOnClickListener {
+          when (getStyle()?.styleURI) {
+            Style.MAPBOX_STREETS -> loadStyle(prepareStyle(Style.SATELLITE_STREETS, bitmap))
+            Style.SATELLITE_STREETS -> loadStyle(prepareStyle(Style.MAPBOX_STREETS, bitmap))
+          }
+        }
+      }
+    }
+  }
+
+  private fun prepareStyle(styleUri: String, bitmap: Bitmap) = style(styleUri) {
+    +image(BLUE_ICON_ID) {
+      bitmap(bitmap)
+    }
+    +geoJsonSource(SOURCE_ID) {
+      featureCollection(FeatureCollection.fromFeatures(pointList))
+    }
+    if (styleUri == Style.SATELLITE_STREETS) {
+      +rasterDemSource(TERRAIN_SOURCE) {
+        url(TERRAIN_URL_TILE_RESOURCE)
+      }
+      +terrain(TERRAIN_SOURCE)
+    }
+    +symbolLayer(LAYER_ID, SOURCE_ID) {
+      iconImage(BLUE_ICON_ID)
+      iconAnchor(IconAnchor.BOTTOM)
+      iconAllowOverlap(false)
+    }
+  }
+
+  override fun onMapLongClick(point: Point): Boolean {
+    val markerId = addMarkerAndReturnId(point)
+    addViewAnnotation(point, markerId)
+    return true
+  }
+
+  override fun onMapClick(point: Point): Boolean {
+    mapboxMap.queryRenderedFeatures(
+      mapboxMap.pixelForCoordinate(point),
+      RenderedQueryOptions(listOf(LAYER_ID), null)
+    ) {
+      if (it.isValue && it.value?.size!! > 0) {
+        it.value?.get(0)?.feature?.let { feature ->
```

Comment:
```suggestion
     getFirstFeatureOrNull?. ...

```
how about small extension?